### PR TITLE
Add recipe for Tinker

### PR DESCRIPTION
**Pull request by njzjz**

### CODE REVIEW

**Comment by njzjz** [comment]
Skip Windows because fftw has not supported Fortran on Windows yet.

@conda-forge/staged-recipes Please review.

File: recipes/tinker/meta.yaml [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+    - cd source
+    - make TINKERDIR=$PREFIX RANLIB=ranlib LIBDIR="-L. -L$PREFIX/lib/linux" F77=$GFORTRAN FFTW_LIBDIR="-L$BUILD_PREFIX/lib"  # [linux]
+    - make TINKERDIR=$PREFIX F77=$GFORTRAN FFTW_LIBDIR="-L$BUILD_PREFIX/lib" LINKFLAGS='$(OPTFLAGS) -static-libgcc -headerpad_max_install_names'  # [osx]
+    - make install TINKERDIR=$PREFIX
+
+requirements:
+  build:
+    - {{ compiler("fortran") }}
+    - make
+    - fftw
```
Comment (scopatz) [inline recipes/tinker/meta.yaml]: Shouldn't this be a host requirements?

Comment (njzjz) [reply recipes/tinker/meta.yaml]: It's my mistake. I've updated it.

File: recipes/tinker/meta.yaml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    - {{ compiler("fortran") }}
+    - make
+  host:
+    - fftw
+  run:
+    - openmp
+
+test:
+  commands:
+    - which alchemy
```
Comment (scopatz) [inline recipes/tinker/meta.yaml]: This test doesn't seem related to the package

Comment (njzjz) [reply recipes/tinker/meta.yaml]: I've added the test commands.

**Comment by njzjz** [comment]
I've added the test commands.

File: recipes/tinker/meta.yaml [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+  commands:
+    - cd test
+    - cp anion.dyn0 anion.dyn
+    - dynamic anion 1000 2.0 0.1 2 300.0
+    - alchemy < anion.dat
+
+about:
+  home: https://dasher.wustl.edu/tinker/
+  license: custom
+  license_file: LICENSE.pdf
```
Comment (chrisburr) [inline recipes/tinker/meta.yaml]: [From the license:](https://github.com/TinkerTools/Tinker/blob/release/LICENSE.pdf)

> The Licensee may not sublicense the use of the Software, or make it available to third parties, without the prior written consent of at least one TCDC-Member.

I think written permission is needed from a TCDC member stating that conda-forge and numFOCUS are allowed distribute this software.
